### PR TITLE
Added support for Apple Silicon. Fixes #1997

### LINKED ISSUE #1997: Gauge CLI not working on Apple Silicon (M1)
**Describe the bug**
Running Gauge does not work on the Apple Silicon. Executing any Gauge command does nothing. No message is printed and no action done. 

I tried node v16.1.0 (npm v7.11.2) and node v14.17.0 (npm v6.14.13) 
Installed gauge via npm and homebrew.

**To Reproduce**
Steps (or project) to reproduce the behavior:
1. Install gauge cli via homebrew or npm 
2. Run the gauge command
3. See no output at all

**Expected behavior**
I would expect that the gauge cli works as on any other OS.

**Screenshots**
image

**Additional context**
 - MacOS 11.3.1 Apple M1 Chip
 - Node Runner
 - IDE: VS Code

### PR DESCRIPTION
**Pull request by Bonitas**
This PR adds support for the Apple Silicon.

I'm not familiar with the different packaging options you are providing. Would really appreciate any hints what else needs to be changed.

### CODE REVIEW

**Comment by sriv** [comment]
Hi @Bonitas - thanks for setting this up.

the DCO check is failing because one or more commits are not signed, please see the notes here: https://github.com/getgauge/gauge/pull/1998/checks?check_run_id=2609216859

As for packaging for macOS, the below options are available:

- `go get` install - nothing to be done, golang's tooling should handle this
- `brew install` - nothing to be done here AFAIK, the homebrew team package and distribute the bottles, so it should work.
- `npm install` - I see you've already added a commit for this.
- `pip install` - same as npm, I see changes to include this.
- `zip install` - It should be addressed since [you've added `arm` to `getPackageArchSuffix()`](https://github.com/getgauge/gauge/pull/1998/files#diff-8dee26221394440174ffc668d5d7e415416a3ceb38f4fa31c8c953792de9e2a9R445-R448) in `make.go`
- codesign - needs testing, and will need to check if github actions has support for this.

tldr - I need to check about codesign. everything else seems to be in place.

**Comment by Bonitas** [comment]
Hi @sriv,

Thanks for your feedback. Would you prefer me to rebase sll commits into one for the DCO? Otherwise I think I'll need to edit the merge commit. Thats the one where I missed the sign.

**Comment by sriv** [comment]
hi @Bonitas 

> Would you prefer me to rebase sll commits into one for the DCO? Otherwise I think I'll need to edit the merge commit. Thats the one where I missed the sign.

Yes, a rebase should be fine in this case.

> Also maybe it's worth to add arm support for linux

Yes, the change should be quite small, but it's the testing that I am worried about too. Let's defer this until someone finds a need for it.

**Comment by Bonitas** [comment]
> hi @Bonitas
> 
> > Would you prefer me to rebase sll commits into one for the DCO? Otherwise I think I'll need to edit the merge commit. Thats the one where I missed the sign.
> 
> Yes, a rebase should be fine in this case.
> 
> > Also maybe it's worth to add arm support for linux
> 
> Yes, the change should be quite small, but it's the testing that I am worried about too. Let's defer this until someone finds a need for it.

Okay rebase is done.

**Review by sriv — DISMISSED** [review]

**Comment by sriv** [comment]
hi @Bonitas - I was hoping to make a release with this change. In order to do so, gauge version needs to be bumped up. I think 1.2.0 would be a good number.

Could you please set the version in `version.go` - see https://github.com/getgauge/gauge/blob/master/CONTRIBUTING.md#bump-up-gauge-version

**Comment by Bonitas** [comment]
> hi @Bonitas - I was hoping to make a release with this change. In order to do so, gauge version needs to be bumped up. I think 1.2.0 would be a good number.
> 
> Could you please set the version in `version.go` - see https://github.com/getgauge/gauge/blob/master/CONTRIBUTING.md#bump-up-gauge-version

version is bumped. Thanks for your patience :)

**Review by sriv — APPROVED** [review]
